examples/llm/src/mosaic_gpt.py

```diff
@@ -86,26 +87,74 @@ class FlashCausalAttention(nn.Module):
     def __init__(self, cfg: DictConfig, device: Optional[str] = None):
         super().__init__()
         try:
-            from flash_attn.flash_attention import FlashMHA  # type: ignore
+            from flash_attn.flash_attention import (  # type: ignore
+                FlashAttention, FlashMHA)
         except ImportError as e:
             raise e
 
-        self.mhsa = FlashMHA(
-            embed_dim=cfg.d_model,
-            num_heads=cfg.n_heads,
-            attention_dropout=cfg.attn_pdrop,
-            bias=True,
-            batch_first=True,
-            causal=True,
-            device=device,
-        )
-        self.mhsa.out_proj._is_residual = True
+        self.attn_qk_ln = cfg.attn_qk_ln
```

Comment:
```suggestion
        self.attn_qk_ln = cfg.get('attn_qk_ln')
```
if we do it this way, we don't need to add it to the config unless we're using it.